server/app/controllers/fusor/api/v21/deployments_controller.rb

```diff
@@ -36,6 +36,13 @@ def create
     end
 
     def update
+      # OpenStack Undercloud attributes should only be set by the undercloud
+      # controller (after it has validated them), never by directly updating
+      # the deployment object.
+      params[:deployment].delete :openstack_undercloud_password
+      params[:deployment].delete :openstack_undercloud_ip_addr
+      params[:deployment].delete :openstack_undercloud_user
+      params[:deployment].delete :openstack_undercloud_user_password
```

Comment:
These 4 lines can stay or be removed. It doesn't hurt anything, but these parameters should always be passed as `null` from the web UI for a PUT call.